src/dashboard/src/main/migrations/0060_create_dataverse_transfer_type.py

```diff
@@ -0,0 +1,582 @@
+
+    # 0af6b163 (Set transfer type: Dataverse)
+    create_ms_chain_link(
+        apps=apps, ms_uuid="0af6b163-5455-4a76-978b-e35cc9ee445f",
+        group="Verify transfer compliance",
+        task_uuid="4d36c35a-0829-4b2d-ba3d-0a30a3e837f9")
+
+    # Create chains for the initiation of Dataverse transfers.
+    create_ms_chain(
+        apps=apps, chain_uuid="35a26b59-dcf3-45ec-b963-ba7bfaa8304f",
+        ms_uuid="246943e4-d203-48e1-ac84-4865520e7c30",
+        chain_description="Dataverse Transfers in Progress")
+    create_ms_chain(
+        apps=apps, chain_uuid="10c00bc8-8fc2-419f-b593-cf5518695186",
+        ms_uuid="fdb12ea6-22aa-46c8-a591-a2bcf5d42e5e",
+        chain_description="Approve Dataverse transfer")
+
+    # Approve Dataverse transfer
+    create_ms_choice(
+        apps=apps, choice_uuid="dc9b59b3-dd5f-4cd6-8e97-ee1d83734c4c",
+        chain_uuid="10c00bc8-8fc2-419f-b593-cf5518695186",
+        link_uuid="246943e4-d203-48e1-ac84-4865520e7c30")
+
+    # Reject Dataverse transfer
+    create_ms_choice(
+        apps=apps, choice_uuid="77bb4993-9f5b-4e60-bbe9-0039a6f5934e",
+        chain_uuid="1b04ec43-055c-43b7-9543-bd03c6a778ba",
+        link_uuid="246943e4-d203-48e1-ac84-4865520e7c30")
+
+    # Create a watched directory which will be where transfers can be
+    # initiated.
+    create_watched_dir(
+        apps=apps, watched_uuid="3901db52-dd1d-4b44-9d86-4285ddc5c022",
+        dir_path="%watchDirectoryPath%activeTransfers/dataverseTransfer",
+        expected_type="f9a3a93b-f184-4048-8072-115ffac06b5d",
+        chain_uuid="35a26b59-dcf3-45ec-b963-ba7bfaa8304f")
+
+    # Create a new link now we have broken the original.
+    # fdb12ea6 (Move to processing directory) connects to:
+    # 0af6b163 (Set transfer type: Dataverse)
+    create_ms_exit_codes(
+        apps=apps, exit_code_uuid="f7e3753c-4df9-43fe-9c32-0d11c511308c",
+        ms_in="fdb12ea6-22aa-46c8-a591-a2bcf5d42e5e",
+        ms_out="0af6b163-5455-4a76-978b-e35cc9ee445f")
+
+    # Create a new link now we have broken the original.
+    # 0af6b163 (Set transfer type: Dataverse) connects to:
+    # 50b67418 (Remove hidden files and directories)
+    create_ms_exit_codes(
+        apps=apps, exit_code_uuid="da46e870-290b-4fd4-8f84-194b9177d8c0",
+        ms_in="0af6b163-5455-4a76-978b-e35cc9ee445f",
+        ms_out="50b67418-cb8d-434d-acc9-4a8324e7fdd2")
+
+
+def data_migration_up(apps, schema_editor):
+    """Run the various groupings of migration functions for the Dataverse
+    specific transfer type.
+    """
+    create_dataverse_transfer_type(apps)
+    create_parse_dataverse_mets_microservice(apps)
+    create_dataverse_unit_variables_and_initial_tasks(apps)
+    create_convert_dataverse_link_pull(apps)
+    create_parse_dataverse_mets_link_pull(apps)
+
+
+def data_migration_down(apps, schema_editor):
+    """Reverse the changes made to the database in order to create a Dataverse
+    transfer type.
+    """
```

Comment:
Nice. Probably was overkill, but to test this I did the following.

First, I removed the new migration and rebuilt the database:

    $ mv src/dashboard/src/main/migrations/0060_create_dataverse_transfer_type.py ~/
    $ make bootstrap-dashboard-db

Then I generated the JSON links workflow as json-links-init.json:

    $ docker-compose ps -q archivematica-dashboard
    5e9dc12e620c8b85ab0b6f6f9371973a8322831487aa3c1c5351321fa475418e
    $ make manage-dashboard ARG=json_links
    /var/archivematica/sharedDirectory/tmp/json-links-kMsuqr.json
    $ docker cp 5e9dc12e620c8b85ab0b6f6f9371973a8322831487aa3c1c5351321fa475418e:/var/archivematica/sharedDirectory/tmp/json-links-kMsuqr.json ./
    $ mv json-links-kMsuqr.json json-links-init.json

Then I returned the Dataverse migration, rebuilt the database again, and
generated the new JSON links workflow as json-links-dataverse.json:

    $ mv ~/0060_create_dataverse_transfer_type.py src/dashboard/src/main/migrations/
    $ make bootstrap-dashboard-db
    $ make manage-dashboard ARG=json_links
    /var/archivematica/sharedDirectory/tmp/json-links-qdxpMI.json
    $ docker cp 5e9dc12e620c8b85ab0b6f6f9371973a8322831487aa3c1c5351321fa475418e:/var/archivematica/sharedDirectory/tmp/json-links-qdxpMI.json ./
    $ mv json-links-qdxpMI.json json-links-dataverse.json

Then I reverted the Dataverse migration and generated the new JSON links
workflow as json-links-reverted.json:

    $ make manage-dashboard ARG='migrate main 0059_siparrange_longblob'
    $ make manage-dashboard ARG=json_links
    /var/archivematica/sharedDirectory/tmp/json-links-rMIh8e.json
    $ docker cp 5e9dc12e620c8b85ab0b6f6f9371973a8322831487aa3c1c5351321fa475418e:/var/archivematica/sharedDirectory/tmp/json-links-rMIh8e.json ./
    $ mv json-links-rMIh8e.json json-links-reverted.json

Showed that the reverted and initial workflow JSON documents are identical:

    $ diff json-links-init.json json-links-reverted.json
    $